cluster/images/etcd/rollback/README.md

```diff
@@ -0,0 +1,45 @@
+# Rollback workflow
+
+Build it in this directory.
+Make sure you have etcd dependency ready. Last time we use etcd v3.0.7.
+```
+$ go build .
+```
+
+
+Run it:
+```
+$ ./rollback2 --data-dir $ETCD_DATA_DIR --ttl 1h
+```
+
+This will rollback KV pairs from v3 into v2.
+If a key was attached to a lease before, it will be created with given TTL (default to 1h).
+
+On success, it will print at the end:
+```
+Finished successfully
+```
+
+Repeat this on all etcd members.
+
+You can do simple check on keys (if any exists):
+```
+etcdctl ls /
+```
+
+Important Note
+------
+
+This tool isn't recommended to use if any problem comes up in etcd3 backend.
+Please report bugs and we will fix it soon.
+
+If it's still preferred to run this tool, please backup all your data beforehand.
+This tool will also back up datadir to same path with ".rollback.backup" suffix.
+
+Caveats:
+- No guarantee on versions.
```

Comment:
Do the versions go back in time? that could be problematic. It's fine if
they jump forward in time.

On Thu, Sep 8, 2016 at 12:28 PM, Xiang Li <notifications@github.com> wrote:

> In cluster/images/etcd/rollback/README.md
> <https://github.com/kubernetes/kubernetes/pull/32161#discussion_r78073405>
> :
>
> > +You can do simple check on keys (if any exists):
> > +```
> > +etcdctl ls /
> > +```
> > +
> > +Important Note
> > +------
> > +
> > +This tool isn't recommended to use if any problem comes up in etcd3 backend.
> > +Please report bugs and we will fix it soon.
> > +
> > +If it's still preferred to run this tool, please backup all your data beforehand.
> > +This tool will also back up datadir to same path with ".rollback.backup" suffix.
> > +
> > +Caveats:
> > +- No guarantee on versions.
>
> do not preserve versions of keys
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/32161/files/205f036b975c79ee466543e5e36737bac5c3e393#r78073405>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAnglnq4uboNXrVvFpGR8rArGXmuDXP8ks5qoGHhgaJpZM4J2UvD>
> .
>